lib/kazan/watcher.ex

```diff
@@ -45,11 +43,17 @@ defmodule Kazan.Watcher do
       {:noreply, state}
     end
   ```
+
+  4. In the case that a "GONE" is received, then this indicates that Kubernetes
+     has sent a 410 error.  In this case the Watcher will automatically terminate
+     and the consumer must clear its cache, reload any cached resources and
+     restart the watcher.
```

Comment:
I'm guessing the fact that we need to trigger restarts on GONE events means that a watcher can't easily be used as part of a supervision tree without some other process explicitly managing the restarts.  Do you have any thoughts about how best to use this as part of a supervision tree that we could document here?

Reply:
In the consumer code that I have I start watchers as a child of a dynamic supervisor with `restart: :transient`.  The watcher will terminate with `{:stop, :normal, state}` when a GONE is received, and so not be restarted by the supervisor.  This should probably be documented.

Another issue that I think may be worth noting is that if a consumer has multiple watches on different resources, when clearing its cache and restarting it should also terminate any old watches by calling `stop_watch/1`.  In my consumer code I do this by storing a list a watchers in the state.  I additionally check that any `Watcher.Event{}` messages received are from the list of watches that I currently have to ensure they are not coming from an old watcher that has not yet terminated or an old message still in the mailbox.